clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/mapreduce/datawriter/jobs/DataWriterMRJob.java

```diff
@@ -214,12 +216,15 @@ private void setupDefaultJobConf(JobConf conf, PushJobSetting pushJobSetting, Ve
         props.getString(ZSTD_COMPRESSION_LEVEL, String.valueOf(Zstd.maxCompressionLevel())));
     conf.setBoolean(ZSTD_DICTIONARY_CREATION_SUCCESS, pushJobSetting.isZstdDictCreationSuccess);
 
-    // We generate a random UUID once, and the tasks of the compute job can use this to build the same producerGUID
-    // deterministically.
-    UUID producerGuid = UUID.randomUUID();
-    conf.setLong(PUSH_JOB_GUID_MOST_SIGNIFICANT_BITS, producerGuid.getMostSignificantBits());
-    conf.setLong(PUSH_JOB_GUID_LEAST_SIGNIFICANT_BITS, producerGuid.getLeastSignificantBits());
-
+    if (pushJobSetting.isMemtableBatchWriteEnabled) {
+      conf.set(GUID_GENERATOR_IMPLEMENTATION, DEFAULT_GUID_GENERATOR_IMPLEMENTATION);
+    } else {
+      // We generate a random UUID once, and the tasks of the compute job can use this to build the same producerGUID
```

Comment:
Could you please add reasoning here why we reuse the same producerGUID for speculative executors from slack thread?